bodhi/server/validators.py

```diff
@@ -163,14 +222,13 @@ def cache_nvrs(request, build):
         return
     if build not in request.buildinfo:
         request.buildinfo[build] = {}
-    name, version, release = get_nvr(build)
 
-    if '' in (name, version, release):
-        raise ValueError
-
-    request.buildinfo[build]['nvr'] = name, version, release
-    # Cram some extra information in there, used later to infer type.
-    request.buildinfo[build]['info'] = request.koji.getBuild(build)
+    # Request info from koji, used to split NVR and determine type
+    # We use Koji's information to get the NVR split, because modules can have dashes in their
+    # stream.
+    kbinfo = request.koji.getBuild(build)
```

Comment:
Might we want to do things the "old way" when they are not ModuleBuilds? I think this is fine too, and is consistent, though it does increase reliance on a network service. Either way is fine with me.

Reply:
This was already there, also we need the buildinfo to determine whether it is a module build.